src/Opaleye/Internal/Column.hs

```diff
@@ -10,12 +15,14 @@ import qualified Opaleye.Internal.HaskellDB.PrimQuery as HPQ
 --
 -- Do not use the 'Show' instance of 'Column'.  It will be deprecated
 -- in version 0.6.
-newtype Column pgType = Column HPQ.PrimExpr deriving Show
+data Column_ colType pgType = Column HPQ.PrimExpr deriving Show
+type Column = Column_ NonNullable
 
 -- | Only used within a 'Column', to indicate that it can be @NULL@.
 -- For example, a 'Column' ('Nullable' @PGText@) can be @NULL@ but a
 -- 'Column' @PGText@ cannot.
 data Nullable a = Nullable
+data NonNullable
```

Comment:
Ok, my mistake then, I assumed since you wanted to lay the groundwork for `PGArray` having a nullability parameter which would not be in use yet, but would be put to use in a future release, it would make sense to also make sense to lay the groundwork for `Column` to have the nullability parameter, which would also not be in use yet, but would be put to use in the future.